src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -691,7 +691,17 @@ public function dumpFile(string $filename, $content)
                 throw new IOException(sprintf('Failed to write file "%s": ', $filename).self::$lastError, 0, null, $filename);
             }
 
-            self::box('chmod', $tmpFile, self::box('fileperms', $filename) ?: 0666 & ~umask());
+            if ($targetFilePermissions = self::box('fileperms', $filename)) {
+                $permissions = $targetFilePermissions;
+
+                if ('\\' === \DIRECTORY_SEPARATOR) {
+                    $permissions |= 0200;
```

Comment:
I'm not sure I understand this: we give write access to $tmpFile, but on L706 we rename that file to $filename, so that the resulting file will always have write access, is that what we want?
Shouldn't we do the chmod in the finally block instead?